File: guild/src/test/java/com/os/operando/guild/EnneadTest.java
package com.os.operando.guild;

import org.junit.Test;

import java.util.*;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.*;

public class EnneadTest {

    @Test
    public void test() {
        Calendar calendar = Calendar.getInstance();
        List<String> strings = Arrays.asList("ennead");
        Set<String> set = Collections.singleton("ennead");
        UUID uuid = UUID.randomUUID();
        StringBuilder sb = new StringBuilder("ennead");
        Ennead<String, Integer, Boolean, Calendar, List<String>, Set<String>, Long, UUID, StringBuilder> ennead =
                Ennead.create("ennead", 0, false, calendar, strings, set, Long.MAX_VALUE, uuid, sb);
        assertThat(ennead.getFirst(), is("ennead"));
        assertThat(ennead.getSecond(), is(0));
        assertThat(ennead.getThird(), is(false));
        assertThat(ennead.getFourth(), is(calendar));
        assertThat(ennead.getFive(), is(strings));
        assertThat(ennead.getSix(), is(set));
        assertThat(ennead.getSeven(), is(Long.MAX_VALUE));
        assertThat(ennead.getEight(), is(uuid));
        assertThat(ennead.getNine(), is(sb));
    }
}

File: guild-kotlin/src/test/kotlin/com/os/operando/guild/kt/EnneadTest.kt
package com.os.operando.guild.kt

import org.hamcrest.Matchers
import org.junit.Assert.*
import org.junit.Test
import java.time.LocalDateTime
import java.util.*

class EnneadTest {

    @Test
    fun test() {
        val now = LocalDateTime.now()
        val uuid = UUID.randomUUID()
        val ennead = 1 to 10L to true to now to "ennead" to listOf("ennead") to Long.MAX_VALUE to 1.0 to uuid
        assertThat(ennead.first, Matchers.`is`(1))
        assertThat(ennead.second, Matchers.`is`(10L))
        assertThat(ennead.third, Matchers.`is`(true))
        assertThat(ennead.fourth, Matchers.`is`(now))
        assertThat(ennead.five, Matchers.`is`("ennead"))
        assertThat(ennead.six, Matchers.`is`(listOf("ennead")))
        assertThat(ennead.seven, Matchers.`is`(Long.MAX_VALUE))
        assertThat(ennead.eight, Matchers.`is`(1.0))
        assertThat(ennead.nine, Matchers.`is`(uuid))
    }

    @Test
    fun testToList() {
        val l = (0 to 1 to 2 to 3 to 4 to 5 to 6 to 7 to 8).toList()
        assertThat(l.size, Matchers.`is`(9))
        assertThat(l[0], Matchers.`is`(0))
        assertThat(l[1], Matchers.`is`(1))
        assertThat(l[2], Matchers.`is`(2))
        assertThat(l[3], Matchers.`is`(3))
        assertThat(l[4], Matchers.`is`(4))
        assertThat(l[5], Matchers.`is`(5))
        assertThat(l[6], Matchers.`is`(6))
        assertThat(l[7], Matchers.`is`(7))
        assertThat(l[8], Matchers.`is`(8))
    }
}